_posts/2021-12-09-kafka-segment-retention.md

```diff
@@ -0,0 +1,259 @@
+---
+layout: post
+title: "Deep dive into Apache Kafka storage internals: segments, rolling and retention"
+date: 2021-12-09
+author: paolo_patierno
+---
+
+Apache Kafka behaves as a commit-log when it comes to deal with storing records.
+Records are appended at the end of each log one after the other and each log is also split in segments
+Segments help with deletion of older records, improving performance, and much more.
+So, the log is a logical sequence of records that's composed of segments (files) and segments store a sub-sequence of records.
+Broker configuration allows you to tweak parameters related to logs. You can use configuration to control the rolling of segments, record retention, and so on.
+
+Not everyone is aware of how these parameters have an impact on broker behavior.
+For instance, they determine how long records are stored and made available to consumers.
+In this blog post, we will dig more into how the log segment configuration and record retention works, in order to help you make better decisions depending on your particular use case.
```

Comment:
```suggestion
In this blog post, we will dig more into how log segmentation and record retention impacts broker performance. When you know more about how it works, you might want to adjust your log configuration. 

> Another aspect of log configuration is enabling compaction. Using keys to identify messages, Kafka compaction keeps the latest message (with the highest offset) for a specific message key. Earlier messages that have the same key are discarded. You can read more on compaction in the Strimzi documentation for [removing log data with cleanup policies](https://strimzi.io/docs/operators/latest/using.html#removing_log_data_with_cleanup_policies).
```

Reply:
^^^I've suggested referring to compaction in a note

Reply:
@PaulRMellor but it's not mentioned or at least not clear that this blog post is valid when policy is "delete". I mean, we are referring to "compact" policy in a note with no context of why we should mention it.

Reply:
I've adjusted the suggestion.